src/expr/impl/src/scalar/array_remove.rs

```diff
@@ -68,11 +68,10 @@ use risingwave_expr::function;
 /// ```
 #[function("array_remove(anyarray, any) -> anyarray")]
 fn array_remove(array: Option<ListRef<'_>>, elem: Option<ScalarRefImpl<'_>>) -> Option<ListValue> {
-    Some(ListValue::new(
-        array?
-            .iter()
-            .filter(|x| x != &elem)
-            .map(|x| x.to_owned_datum())
-            .collect(),
-    ))
+    let array = array?;
+    let mut builder = array.data_type().create_array_builder(array.len());
+    for val in array.iter().filter(|x| x != &elem) {
+        builder.append(val);
+    }
+    Some(ListValue::new(builder.finish()))
```

Comment:
It seems like this pattern happens many times (create builder, iter & append, finish), why `from_iter` doesn't work here?

Reply:
👍 It needs an additional data type parameter.